mmdet/core/evaluation/eval_hooks.py

```diff
@@ -53,6 +53,8 @@ def _do_evaluate(self, runner):
             return
 
         from mmdet.apis import single_gpu_test
+        # assert when dataloader is empty, in case that Error Occurs when running train.py with the "image" field of json file (val2017.json) set empty
```

Comment:
Not all task uses json as an annotation file. Try to make the error message and comments be more specific.